wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/UserDevfileManager.java

```diff
@@ -0,0 +1,140 @@
+      throws ServerException, NotFoundException, ConflictException {
+    requireNonNull(devfile, "Required non-null devfile");
+    UserDevfileImpl result =
+        userDevfileDao.create(
+            new UserDevfileImpl(NameGenerator.generate("usrdevfile", 16), devfile));
+    LOG.info(
+        "UserDevfile '{}' with id '{}' created by user '{}'",
+        result.getName(),
+        result.getId(),
+        EnvironmentContext.getCurrent().getSubject().getUserName());
+    eventService.publish(new DevfileCreatedEvent(result));
+    return result;
+  }
+
+  /**
+   * Gets UserDevfile by given id.
+   *
+   * @param id userdevfile identifier
+   * @return userdevfile instance
+   * @throws NullPointerException when {@code id} is null
+   * @throws NotFoundException when userdevfile with given id not found
+   * @throws ServerException when any server errors occurs
+   */
+  public UserDevfile getById(String id) throws NotFoundException, ServerException {
+    requireNonNull(id);
+    return userDevfileDao.getById(id);
+  }
+
+  /**
+   * Updates an existing user devfile in accordance to the new configuration.
+   *
+   * <p>Note: Replace strategy is used for user devfile update, it means that existing devfile data
+   * will be replaced with given {@code update}.
+   *
+   * @param update user devfile update
+   * @return updated user devfile
+   * @throws NullPointerException when {@code update} is null
+   * @throws ConflictException when any conflict occurs.
+   * @throws NotFoundException when user devfile with given id not found
+   * @throws ServerException when any server error occurs
+   */
+  public UserDevfileImpl updateUserDevfile(UserDevfile update)
+      throws ConflictException, NotFoundException, ServerException {
+    requireNonNull(update);
+    UserDevfileImpl result = userDevfileDao.update(new UserDevfileImpl(update));
+    LOG.info(
+        "UserDevfile '{}' with id '{}' update by user '{}'",
+        result.getName(),
+        result.getId(),
+        EnvironmentContext.getCurrent().getSubject().getUserName());
+    eventService.publish(new DevfileUpdatedEvent(result));
+    return result;
+  }
+
+  /**
+   * Removes stored {@link UserDevfile} by given id.
+   *
+   * @param id user devfile identifier
+   * @throws NullPointerException when {@code id} is null
+   * @throws ServerException when any server errors occurs
+   */
+  public void removeUserDevfile(String id) throws ServerException {
+    requireNonNull(id);
+    userDevfileDao.remove(id);
+    LOG.info(
+        "UserDevfile with id '{}' removed by user '{}'",
+        id,
+        EnvironmentContext.getCurrent().getSubject().getUserName());
+    eventService.publish(new DevfileDeletedEvent(id));
+  }
+
+  /**
+   * Gets list of devfiles. Parameters, returned values and possible exceptions are the same as in
+   * UserDevfileDao#getDevfiles(String, int, int, List, List)}.
+   */
+  public Page<UserDevfileImpl> getUserDevfiles(
+      String user,
```

Comment:
```suggestion
      String userId,
```
^ to make it clearer that it's not username